concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/10 Appearance and Editing/20 Editor Customization.md

```diff
@@ -0,0 +1,169 @@
+DataGrid generates editors based on the **columns[]**.[dataType](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataType) property. The editors are generated with the default options. This section demonstrates how to change the default editor settings, implement any of built-in editors, or create your custom editor.
+
+If you want to change the predefined editor properties, specify the **columns[].**[editorOptions](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editorOptions) property, that applies to header filter editors as well. Do not specify the **onValueChanged** handler in this property, because the internal DataGrid logic overrides it. The **columns[].formItem.**[editorOptions](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/#editorOptions) property applies to form items only. 
+
+[note]Do not specify the **template** property in **formItem.editorOptions** or **columns[].editorOptions**. To implement a custom template, use the [editCellTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editCellTemplate) property.
+
+#include common-demobutton with {
+    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/CustomEditors/"
+}
+
+You can override the **onValueChanged** handler of the predefined editor and change the predefined editor type. For this purpose, handle the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditorPreparing) event. Use handler parameters to determine the editor location and the **dataField** to which the editor is bound. Set the **editorName** parameter to the required DevExtreme editor component and the **editorOptions** parameter to the editor configuration object.
```

Comment:
> Use handler parameters to determine the editor location and the **dataField** to which the editor is bound. Set the **editorName** parameter to the required DevExtreme editor component and the **editorOptions** parameter to the editor configuration object.

These sentences look somewhat confusing. At first, you speak of handler parameters in general when it comes to editor location, and then start to name particular parameters all of the sudden. The point is, I do not understand why editor location gets special treatment in this case.

If you wan to explain everything in detail, I would recommend you to make a list of parameters along with the corresponding explanations.  Every time you iterate more than two things in plain text, consider creating a list instead.

As I mentioned earlier on, this paragraph looks somewhat confusing in its current state.

Reply:
I agree with your suggestions! I think, we can specify only the following information: You can override the **onValueChanged** handler of the predefined editor and change the predefined editor type. For this purpose, use the **onEditorPreparing** property.

Reply:
Therefore, you believe that it's not worth going into greater detail, am I right? That's ultimately your choice to make -- I have no objections here.

I believe we can stop right here:

```suggestion
You can override the **onValueChanged** handler of the predefined editor and change its type. To accomplish this goal, handle the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditorPreparing) event.
```

Reply:
Yes, there is a code snippet with details.